robolectric/src/test/java/org/robolectric/android/internal/AndroidTestEnvironmentTest.java

```diff
@@ -111,14 +117,49 @@ public void setUpApplicationState_setsMainThread_onAnotherThread() throws Interr
    * in Robolectric via {@link BouncyCastleProvider}.
    */
   @Test
-  public void ensureBouncyCastleInstalled() throws GeneralSecurityException {
+  @ConscryptMode(ON)
+  public void testWhenConscryptModeOn_ConscryptInstalled()
+      throws CertificateException, NoSuchAlgorithmException {
 
     bootstrapWrapper.callSetUpApplicationState();
+    CertificateFactory factory = CertificateFactory.getInstance("X.509");
+    assertThat(factory.getProvider().getName()).isEqualTo("Conscrypt");
 
+    MessageDigest digest = MessageDigest.getInstance("SHA256");
+    assertThat(digest.getProvider().getName()).isEqualTo("Conscrypt");
+  }
+
+  @Test
+  @ConscryptMode(ON)
+  public void testWhenConscryptModeOn_BouncyCastleInstalled() throws GeneralSecurityException {
+
+    bootstrapWrapper.callSetUpApplicationState();
+    Cipher aesCipher = Cipher.getInstance("RSA/None/OAEPWithSHA-256AndMGF1Padding");
+    assertThat(aesCipher.getProvider().getName()).isEqualTo(BouncyCastleProvider.PROVIDER_NAME);
+  }
+
+  @Test
+  @ConscryptMode(OFF)
+  public void testWhenConscryptModeOff_ConscryptInstalled()
+      throws CertificateException, NoSuchAlgorithmException {
+
+    bootstrapWrapper.callSetUpApplicationState();
+    CertificateFactory factory = CertificateFactory.getInstance("X.509");
+    assertThat(factory.getProvider().getName()).isNotEqualTo("Conscrypt");
+
+    MessageDigest digest = MessageDigest.getInstance("SHA256");
+    assertThat(digest.getProvider().getName()).isNotEqualTo("Conscrypt");
+  }
+
+  @Test
+  @ConscryptMode(OFF)
+  public void testWhenConscryptModeOff_BouncyCastleInstalled() throws GeneralSecurityException {
+
+    bootstrapWrapper.callSetUpApplicationState();
     MessageDigest digest = MessageDigest.getInstance("SHA256");
     assertThat(digest.getProvider().getName()).isEqualTo(BouncyCastleProvider.PROVIDER_NAME);
 
-    Cipher aesCipher = Cipher.getInstance("AES/CBC/PKCS7Padding");
+    Cipher aesCipher = Cipher.getInstance("RSA/None/OAEPWithSHA-256AndMGF1Padding");
```

Comment:
What happens if this is changed back to "AES/CBC/PKCS7Padding"?

Reply:
It is working.

Reply:
Previously there were some tests failed, so we changed it.